pkg/compiler/compiler.go

```diff
@@ -26,6 +27,12 @@ type Options struct {
 
 	// The name of the output for debug info.
 	DebugInfo string
+
+	// The name of the output for application binary interface info.
+	ABIInfo string
+
+	// Contract metadata.
+	ContractDetails *request.ContractDetails
```

Comment:
Please move `ContractDetails` out of `rpc/request` to `smartcontract`, importing anything RPC-related into the compiler is just weird.